src/ViaPhone.php

```diff
@@ -22,120 +22,48 @@ class ViaPhone
 	const CALL_STATE_NOT_ANSWERED = 'not_answered';
 	const CALL_STATE_REFUSED = 'refused';
 
-	/** @var string */
-	protected $url = 'https://api.viaphoneapp.com/v2/';
+	protected string $url = 'https://api.viaphoneapp.com/v2/';
+	protected string $lang = 'en';
+	protected Client $client;
+	protected array $headers;
 
-	/** @var string */
-	protected $secret;
 
-	/** @var string */
-	protected $lang = 'en';
-
-	/**
-	 * ViaPhone constructor.
-	 *
-	 * @param string $apiKey
-	 */
-	public function __construct($apiKey)
+	public function __construct(string $apiKey)
 	{
-		$this->secret = $apiKey;
+		$this->client = new Client(['base_uri' => $this->url]);
+		$this->headers = [
+			'X-API-Key' => $apiKey,
+			'Accept-Language', $this->lang,
+		];
 	}
 
-	/**
-	 * @param string $path
-	 * @param array $queryParams
-	 * @return string
-	 */
-	protected function getUrl(string $path, array $queryParams = [])
-	{
-		$url = $this->url . $path;
-
-		if ($queryParams) {
-			$url = new Url($url);
-			$url->setQuery($queryParams);
-		}
-
-		return (string)$url;
-	}
 
-	/**
-	 * @param string $url
-	 * @param array $data
-	 * @param string $method
-	 * @return bool|object|string
-	 * @throws \Exception
-	 */
-	protected function request(string $url, array $data = [], string $method = IRequest::GET)
+	protected function request(string $url, string $method, array $data = [])
```

Comment:
mozna bychom data mohli pojmenovat jako $paramsOrData?

Reply:
Neviem no, niecoOrNieco sa mi obecne nepaci. Mozme pridat o jeden parameter naviac a mat `$parameters` aj `$data` (to sa mi moc nepaci lebo je to zavisle na metode) alebo to request mozeme pridat funkcie `get`, `post`,`patch`, ktore budu volat metodu `request.

Reply:
ne tak je to interni zalezitost takze je zbytecne to komplikovat dalsimi internimi funkcemi... a vic parametru urcite taky nechceme... nechme tam tedy data...